src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -36,9 +36,17 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
 				TypeDefinition attributeType;
 
 				string internalAttribute = GetAttribute (iterator.Current, "internal");
-				if (internalAttribute == "RemoveAttributeInstances" && provider != null && !Annotations.IsMarked (provider)) {
-					IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
-					Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
+				if (internalAttribute != String.Empty) {
+					if (internalAttribute == "RemoveAttributeInstances") {
+						if (provider != null) {
+							if (!Annotations.IsMarked (provider)) {
+								IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
+								Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
+							}
+						} else
+							Context.LogWarning ($"Internal attribute 'RemoveAttributeInstances' can only be used on a type, but is being used on '<{nav.Name}>'", 2048, _xmlDocumentLocation);
```

Comment:
I think we should pass the provider always and print out the provider in the warning message. That will point exactly at the spot where the problem is. Currently it will only say "method", but not which one. (Unlike roslyn, we can't rely on providing context via source/line info - although we could add that and then we could).

Reply:
I changed to use the providers for everything, in the specific case of method params I still print the method as the provider since I cannot pass an enumeration of method params to the function.